src/components/study-container.js

```diff
@@ -472,28 +484,53 @@ export function StudyContainer({ view, onChangeTab }) {
                 dispatch(setCurrentTreeNode(ModelFirstSelectedNode));
             })
             .catch((error) => {
-                if (error.status === 404) {
-                    snackError({
-                        headerId: 'StudyUnrecoverableStateRecreate',
-                    });
-                } else {
-                    snackError({
-                        messageTxt: error.message,
-                        headerId: 'NetworkModificationTreeLoadError',
-                    });
-                }
+                snackError({
+                    messageTxt: error.message,
+                    headerId: 'NetworkModificationTreeLoadError',
+                });
             })
             .finally(() =>
                 console.debug('Network modification tree loading finished')
             );
         // Note: studyUuid and dispatch don't change
     }, [studyUuid, dispatch, snackError, snackWarning]);
 
+    const checkNetworkExistenceAndReimportIfNotFound = useCallback(
+        (successCallback) => {
+            fetchNetworkExistence(studyUuid, true)
+                .then(() => {
+                    successCallback && successCallback();
+                    setIsStudyNetworkFound(true);
+                    loadTree();
+                })
+                .catch((error) => {
+                    //TODO: improve the way we catch this error
+                    if (error.message.endsWith('BROKEN_STUDY')) {
+                        setIsStudyNetworkFound(false);
+                        setIsImportStudyDialogDisplayed(true);
+                        snackError({
+                            headerId: 'StudyUnrecoverableState',
+                        });
+                    } else {
```

Comment:
That means in every other cases, we are sure we can reimport it ?
For me later on, we will be able to know if it's archived or not, and reimport it (only in this case), without having to manage an exception if we want to.
Maybe let a comment in this way ?

Reply:
If possible, meanwhile we wait for the archived studies, we should do differently, like 
if ('MISSING_STUDY') {
... reimport study code ...
}
which will be easily changed into 
if ('ARCHIVED_STUDY') {

Reply:
the future workflow might be very different from what we have now, I don't think we should assume we will only have to replace it like that


I agree with this "else" not being accurate all the time though
If the server is down, we should not display "Study is being reimported"
I will make something more accurate for that case